src/main/java/org/traccar/protocol/TotemProtocolSmsEncoder.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2015 Irving Gonzalez
+ * Copyright 2015 - 2019 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.protocol;
+
+import org.traccar.StringProtocolEncoder;
+import org.traccar.model.Command;
+import org.traccar.Protocol;
+
+public class TotemProtocolSmsEncoder extends StringProtocolEncoder {
+
+    public TotemProtocolSmsEncoder(Protocol protocol) {
+        super(protocol);
+    }
+
+    @Override
+    protected Object encodeCommand(Command command) {
+
+        initDevicePassword(command, "000000");
```

Comment:
Why do we need this?

Reply:
If you mean the initDevicePassword - it sets the password, "000000" is the default password for this device if none is specified. If you mean the whole SmsEncoder, it's because the format of SMS Commands and GPRS commands is a bit different (the "core" is the same, but start and end are different - example checksum is used with GPRS commands but not with GPS ones)

Reply:
Where do you use this password?

Reply:
In formatCommand we get this password from the command for both protocols

Reply:
OK, this is not good. We need to pass the pass word to the static method as an argument.

Reply:
Wait. Ignore what I just said. We are passing it.